src/core/transports/transport_manager.ts

```diff
@@ -47,6 +53,24 @@ export default class TransportManager {
 
   /** Takes one life from the transport. */
   reportDeath() {
+    if (this.successfulConnectionTimer) {
+      this.successfulConnectionTimer.ensureAborted();
+      this.successfulConnectionTimer = null;
+    }
     this.livesLeft -= 1;
   }
+
+  reportConnection() {
```

Comment:
Can/shoud we report success without timers by listening for `connection_established`?

Reply:
Hello 👋 , I was author of the issue, that leads to this PR. 
New to the codebase, but agreed, can we somehow just report the successful connection in the method below and not use any timers? Or it will break the flow somehow? @leesio